covid_graphs/covid_graphs/scripts/rest.py

```diff
@@ -0,0 +1,105 @@
+        # vytvor strukturu pre predikcie a krajiny
+        # vytvor strukturu pre data
+        self.available_predictions = Rest._create_available_predictions_by_date()
+        self.country_reports = Rest._create_country_reports(data_dir)
+        predictions, country_reports_active = Rest._create_predictions(self.country_reports)
+        self.predictions = predictions
+        self.country_reports_active = country_reports_active
+
+    @staticmethod
+    def _create_available_predictions_by_date():
+        result = {}
+        for x in prediction_db.get_prediction_events():
+            result[x.name] = {
+                "label": x.date,
+                "countries": [p.country for p in prediction_db.predictions_for_event(x)],
+            }
+        return result
+
+    @staticmethod
+    def _create_country_reports(data_dir: Path):
+        country_reports = {}
+        for country_short_name in prediction_db.get_countries():
+            if (data_dir / f"{country_short_name}.data").is_file():
+                country_report = create_report(
+                    data_dir / f"{country_short_name}.data", country_short_name
+                )
+                country_reports[country_report.short_name] = country_report
+
+        return country_reports
+
+    @staticmethod
+    def _create_predictions(country_reports):
+        predictions = []
+        country_reports_active = {}
+        for country in prediction_db.get_countries():
+            country_predictions = prediction_db.predictions_for_country(country)
+            country_report = country_reports[country]
+            graph = CountryGraph(report=country_report, country_predictions=country_predictions)
+            country_reports_active[country] = {
+                "type": "cumulative_active",
+                "date_list": graph.cropped_dates,
+                "values": graph.cropped_cumulative_active.tolist(),
+                "short_name": country_report.short_name,
+                "long_name": country_report.long_name,
+            }
+            for prediction, curve in zip(country_predictions, graph.curves):
+                xs, ys = curve.generate_trace(graph.display_until)
+                idx_max = ys.argmax()
+                predictions.append(
+                    {
+                        "type": "prediction",
+                        "date_list": xs,
+                        "values": ys.tolist(),
+                        "description": curve.label,
+                        "short_name": country_report.short_name,
+                        "long_name": country_report.long_name,
+                        "max_value_date": xs[idx_max],
+                        "max_value": ys.max(),
+                        "date_name": prediction.prediction_event.name,
+                        "date": prediction.prediction_event.date,
+                    }
+                )
+
+        return predictions, country_reports_active
+
+    def get_app(self):
+        return self
```

Comment:
What is this useful for?

Reply:
Keeping same convention as is in CountryDashboard

Reply:
I don't think a lot of thought went into naming of CountryDashboard methods :) (Honestly, there is already a confusion already between flask app and dash app so... yeah.)